<launch>

    <!-- Types of Markers Are:
           'MESH' - (Currently Disabled) A Mesh Resource must be Specified
           'QUAD' - (Currently Disabled) A Flat Square Object
           'DISK' - (Currently Disabled) A Flat Circular Object
           'VALVE' - An Object designed for the DRC Valve Turning Task 

         Choices of Color Are:
           'RED' - A Red Marker
    -->

    <param name="/localization_marker_server/marker/type" 
           value="VALVE" 
           type="string" />

    <param name="/localization_marker_server/marker/mesh_resource" 
           value="package://localization_markers/resources/torus.dae" 
           type="string" />

    <param name="/localization_marker_server/marker/size_meters" 
           value=".25" 
           type="double" />

    <param name="/localization_marker_server/marker/color" 
           value="RED" 
           type="string" />

<!-- ---------------------------------------------------------------------- -->

    <!-- Choices for Bounding Box Are:
           'CUBE' - A Cube Box, Size must be specified!
           'SPHERE' - A Sphere, Size must be specified!

         Choices of Color Are:
           'BLUE' - A Blue Bounding Box
    -->

    <param name="/localization_marker_server/bounding_box/type" 
           value="CUBE"
           type="string" />

    <param name="/localization_marker_server/bounding_box/size" 
           value=".30" 
           type="double" />

    <param name="/localization_marker_server/bounding_box/color" 
           value="BLUE" 
           type="string" />

<!-- ---------------------------------------------------------------------- -->

    <!-- Choices for Control Are:
           'TRANS' - Translation Control of the Marker
           'ROT'   - Rotation Control of the Marker
           '6DOF'  - Full 6 DOF Control of the Marker
    -->

    <param name="/localization_marker_server/control" 
           value="6DOF" 
           type="string" />

<!-- ---------------------------------------------------------------------- -->

    <!-- These are Paramters for the Point Clouds, ALL are mandatory
         Further Documentation can be found at www.WEBSITE.com -->

    <param name="/localization_marker_server/target_cloud/topic" 
           value="/kinect_head/depth_registered/points" 
           type="string" />

    <param name="/localization_marker_server/target_cloud/voxel" 
           value="false" 
           type="bool" />

    <param name="/localization_marker_server/input_cloud/resource" 
           value="/" 
           type="string" />

    <param name="/localization_marker_server/input_cloud/discretization" 
           value="100" 
           type="int" />

<!-- ---------------------------------------------------------------------- -->

    <!-- These are Paramters for the Iterative Closest Point, ALL are mandatory
         Further Documentation can be found at www.WEBSITE.com -->


    <param name="/localization_marker_server/icp/maxCoorespondanceDistance" 
           value=".25" 
           type="double" />

    <param name="/localization_marker_server/icp/maxInterations" 
           value="500" 
           type="int" />

<!-- ---------------------------------------------------------------------- -->

    <!-- These are Paramters for the Debugging The System, ALL are mandatory
         Further Documentation can be found at www.WEBSITE.com -->

    <param name="/localization_marker_server/debug/input_cloud/pre_ICP/show" 
           value="true" 
           type="bool" />

    <param name="/localization_marker_server/debug/input_cloud/pre_ICP/topic" 
           value="/localization_marker_server/debug/input_cloud/pre_icp" 
           type="string" />

    <param name="/localization_marker_server/debug/input_cloud/post_ICP/show" 
           value="true" 
           type="bool" />

    <param name="/localization_marker_server/debug/input_cloud/post_ICP/topic" 
           value="/localization_marker_server/debug/input_cloud/post_icp" 
           type="string" />








<!-- ---------------------------------------------------------------------- -->
<!--                DO NOT CHANGE ANYTHING AFTER THIS POINT                 -->
<!-- ---------------------------------------------------------------------- -->

    <param name="/localization_marker_server/hasChanged" 
           value="false" 
           type="bool" />

    <node name="localization_marker_server_drc" 
          pkg="localization_markers" 
          type="localization_marker_server_drc" />

</launch>
